### ISSUE TITLE
TypeError: 'tuple' object does not support item assignment

### ISSUE THREAD

**Issue by glenharris** [Breina/ha-artnet-led#49]
Hi All,

Is there a potential issue with tuple vs array in the class DmxRGBW (light.py).  The reason I ask is that when I switch between turning it on use hue/saturation and RGBW, I get errors (TypeError: 'tuple' object does not support item assignment, light.py, line 800).  I am a python novice, but when I look at that code it seems that some of the class is expecting `_vals` to be an array ('`[]`'), and some to be a tuple ('`()`').  Is that plausible?

 The following minimal pyscript will reproduce the problem (I think, and assuming light.s101 is an artnet rgbw led):
```
def test_artnet():
    entity = 'light.s101'
    rgbw = {
        "rgbw_color": (0,0,255,0)
    }
    hs = {
        "color_temp": 4500,
        "brightness": 70,
    }
    light.turn_on(entity_id=entity, **rgbw)
    light.turn_on(entity_id=entity, **hs)
```

If I just call using hue/saturation, (i.e. without the preceding rgbw_color call), then there is no problem (I think because `_vals` is initialised with an array).

Apologies if I am on the wrong track.

TypeError: 'tuple' object does not support item assignment
2023-11-26 14:18:53.858 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall light.turn_on (c:01HG4KAFX1ZNRGG472PXTTX4ZJ): entity_id=['light.s101'], params=brightness=70, color_temp=4504, color_temp_kelvin=222>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1928, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1950, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 226, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 811, in entity_service_call
    future.result()  # pop exception if have
    ^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1034, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 851, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 582, in async_handle_light_on_service
    await light.async_turn_on(**filter_turn_on_params(light, params))
  File "/config/custom_components/artnet_led/light.py", line 800, in async_turn_on
    self._vals[0:3] = list(color_util.color_hs_to_RGB(hue, sat))

**Comment by Breina** [comment]
Hey man,

I can reproduce, so it's us. Will fix.

**Comment by Breina** [comment]
Fixed in v0.1.19

Please test. :)

**Comment by glenharris** [comment]
:+1: 

Thanks @Breina, that seems to have fixed it.

By the way, thanks for all the hard work you are putting into this - it is very useful, and seems like a quality piece of code.  :smile:
